.travis.yml

```diff
@@ -17,47 +6,92 @@ env:
   - secure: k9mPDnNDPA2LKE+u+Z8hDNAdRxD0WiOqbxHRKArX7TuQZRUhznvtMcMYtSv5aF2iL8c8M9814Et2mXoJOage/Wcup8WVZeA403cS7qogt3/gJLZ8Tz1ZkOyt8d7fs6xAYIowqM0qzoqRJ8V9TcjFVmJ3p3hZjzbl/meHF4hsnHYYP2XLzAKyBqepTjCIriD/zOk+P4uF1U7XwCVwfqNdsRjw8VmI/uEZyYX1ekscxf4AJM67kDmYeojcPf/R8h6ufEcTExL7tweQZ6ypXQVjCGJIalb1NsPhfRFJ9f/8Ti4r3Gsq/2KO4JKKW8pu5sf1rx3QZD5jxK57BDXd/cqV94ejU4mBo9cYHxQZ73A2aqMa0PkXBuAF24DpFoRkvamb9aDk30qOxrTmMSZ8Oee1KUlYhtsOtt3Qhm7qjvS6PUoTYJJqKXLK0VsjCyynt20RUH+6p94uNEgrETxDCIVYzS4Eethx1N6GqxKKs1V8cKopwC3C+kJLsYCtVBR/sDxS1IcbnJ7nojNDFr6O6ize5vH7arUiuBmD23ojPIXqL/VFew4RFlD9FsObrVQOqXRneIDOCxNuLBr5+Vsg3/UkMQqZVvUtENszvpPAr5mac92vLLCcwZZoUai7Zj17XZtDi83wyJoHgQboBJbDAv69TQZFZXTNKUhSHiua1ZsYcNo=
 
-# Do not choose a language; we provide our own build tools.
-language: generic
-
-services:
-  - postgresql
-  - docker
-
-# Caching so the next build will be fast too.
-cache:
-  timeout: 4000
-  directories:
-  - "$HOME/.stack"
-  - ".stack-work"
-
-# Ensure necessary system libraries are present
-addons:
-  apt:
-    packages:
-      - libgmp-dev
-
-before_install:
-# Decrypt the ssh key that will be used for deployments 
-- openssl aes-256-cbc -K $encrypted_8fe1f88deb69_key -iv $encrypted_8fe1f88deb69_iv
-  -in deploy/deployment.pem.enc -out ~/.ssh/deployment.pem -d
-# Download and unpack the stack executable
-- mkdir -p ~/.local/bin
-- export PATH=$HOME/.local/bin:$PATH
-- wget -qO- https://get.haskellstack.org/ | sh
-
-install:
-# Build dependencies
-- stack --no-terminal --install-ghc test --only-dependencies
-
-script:
-# Build the package, its tests, and its docs and run the tests
-- stack clean
-- stack --no-terminal build --copy-bins
-- stack ls dependencies --license | ./license_check.py
-- psql -c 'CREATE DATABASE testsupplychainserver;' -U postgres
-- psql -c 'CREATE DATABASE testbusinessregistry;' -U postgres
-- stack exec supplyChainServer -- --init-db -c 'dbname=testsupplychainserver'
-- echo 'YES' | stack exec businessRegistry -- initdb -c 'dbname=testbusinessregistry'
-- stack test --ta -j1
-- if [ "$TRAVIS_BRANCH" == "master" ]; then source deploy/deploy.sh; fi
+stages:
+  - "Build"
+  - name: "Deploy"
+    if: branch = master
+    
+jobs:
+  include:
+    - name: "Web"
+      stage: "Build"
+      language: node_js
+      node_js:
+        - "11"
+      cache: npm
+      services:
+        - docker
+      env:
+        DOCKER_TAG: ${DOCKER_REPO}/mirza-web:$TRAVIS_COMMIT
+      before_install:
+        - cd ./web
+      install:
+        - npm install
+      script:
+        - ./build.sh
+        - |
+          if [ "$TRAVIS_BRANCH" == "master" ]; then
```

Comment:
So it looks like Travis-ci will try to run this on pull requests to master (it pulls the code in to master and runs the CI process for each PR)

This means that just checking if the branch is master or not isn't enough to stop deployments from happening!

According to some googling, we can fix this if we add a check for:
`if [[ "${TRAVIS_PULL_REQUEST}" = "false" ]]`